### PR TITLE
Add balance check to Approve Budget and Fund

### LINKED ISSUE #2116: Check eth balance before starting deposit process

### PR DESCRIPTION
**Pull request by lalexgap**
Fixes #2116 

Adds a balance check to the approve budget and fund workflow and removes it from the application.

The workflow waits until the user deposits enough funds before proceeding.

image

### CODE REVIEW

File: packages/xstate-wallet/src/chain.ts [hunk 1]
```diff
@@ -199,11 +200,18 @@ export class FakeChain implements Chain {
 
     return merge(first, updates);
   }
-
+  public balanceUpdatedFeed(): Observable<BigNumber> {
```
Comment (kerzhner) [inline packages/xstate-wallet/src/chain.ts:203]: My understanding of having two implementations of the `balanceUpdatedFeed` is that the address parameter is optional. Is it cleaner to have one implementation with an optional parameter?
```suggestion
  public balanceUpdatedFeed(address?: string): Observable<BigNumber> {
```

File: packages/xstate-wallet/src/ui/approve-budget-and-fund-workflow.tsx [hunk 2]
```diff
@@ -29,6 +29,31 @@ export const ApproveBudgetAndFund = (props: Props) => {
     };
   }, [current]);
 
+  const waitForSufficientFundsInit = (
+    <Flex alignItems="center" flexDirection="column">
+      <Heading>App Budget</Heading>
+
+      <Text textAlign="center">Querying blockchain</Text>
```
Comment (kerzhner) [inline packages/xstate-wallet/src/ui/approve-budget-and-fund-workflow.tsx:36]:
```suggestion
      <Text textAlign="center">Checking if your Metamask account has sufficient ETH</Text>
```

File: packages/xstate-wallet/src/ui/approve-budget-and-fund-workflow.tsx [hunk 3]
```diff
@@ -29,6 +29,31 @@ export const ApproveBudgetAndFund = (props: Props) => {
...
+  const waitForSufficientFunds = (
+    <Flex alignItems="left" flexDirection="column">
+      <Heading textAlign="center" mb={2}>
+        App Budget
+      </Heading>
+      <Text pb={3} fontSize={1}>
+        You don&#39;t have enough ETH in your wallet!
+      </Text>
+      <Text pb={3} fontSize={1}>
+        You&#39;ll need at least {utils.formatEther(playerAmount)} ETH in your wallet to fund the
```
Comment (kerzhner) [inline packages/xstate-wallet/src/ui/approve-budget-and-fund-workflow.tsx:49]:
```suggestion
        You&#39;ll need at least {utils.formatEther(playerAmount)} ETH in your Metamask wallet to fund the
```

File: packages/xstate-wallet/src/chain.ts (hunk 1 above)
Comment (lalexgap) [reply packages/xstate-wallet/src/chain.ts:203]: There are actually two classes in here `FakeChain` and `ChainWatcher` which both implement the `Chain` interface. So I needed to add the method to both classes.

Outside the scope of this PR but we should probably move `FakeChain` out of `chain.ts`

Comment (kerzhner) [reply packages/xstate-wallet/src/chain.ts:203]: Ah that makes sense!

**Pull request merged**